modules/collect/src/main/java/com/opengamma/collect/io/IniFile.java

```diff
@@ -84,6 +108,56 @@ public static IniFile of(CharSource source) {
     }
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Returns a single INI file that is the chained combination of the inputs.
+   * <p>
+   * A chained INI file must have one special section, 'chain', with three properties.
+   * The first is 'priority', an integer, which allows the input files to be sorted.
+   * The second is 'nextInChain', a boolean, which determines where to chain on to the next file or not.
+   * The third is 'removedSections', a list of sections to be removed.
+   * All entries from the highest priority file are included in the result, plus
+   * any entries that have not been overridden or removed in chained files.
+   * 
+   * @param sources  the INI file sources to read
+   * @return the combined chained INI file
+   * @throws UncheckedIOException if an IO error occurs
+   * @throws IllegalArgumentException if the configuration is invalid
+   */
+  public static IniFile ofChained(Stream<CharSource> sources) {
+    ArgChecker.notNull(sources, "sources");
+    List<IniFile> files = sources
+        .map(IniFile::of)
+        .sorted(IniFile::sortByReversePriority)
+        .collect(toList());
+    // combine files, based on chain flag
+    Map<String, PropertySet> builder = new LinkedHashMap<>();
+    for (IniFile file : files) {
+      // remove everything from lower priority files if not chaining
+      if (Boolean.parseBoolean(file.getSection(CHAIN_SECTION).getValue(NEXT_IN_CHAIN)) == false) {
+        builder.clear();
+      } else {
+        // remove sections from lower priority files
+        builder.keySet().removeAll(file.getSection(CHAIN_SECTION).getValueList(REMOVED_SECTIONS));
+      }
+      // add entries, replacing existing data
+      for (String sectionName : file.asMap().keySet()) {
+        if (!sectionName.equals(CHAIN_SECTION)) {
+          builder.merge(sectionName, file.getSection(sectionName), PropertySet::combineWith);
+        }
+      }
+    }
+    return new IniFile(ImmutableMap.copyOf(builder));
+  }
+
+  // sort by priority, lowest first
+  private static int sortByReversePriority(IniFile a, IniFile b) {
```

Comment:
I think it would be better to call this ``compareByReversePriority`` as it doesn't actually do any sorting.